pkg/service/recordingservice.go

```diff
@@ -0,0 +1,89 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/livekit/protocol/utils"
+	"google.golang.org/protobuf/proto"
+
+	livekit "github.com/livekit/livekit-server/proto"
+)
+
+type RecordingService struct {
+	rc *redis.Client
+}
+
+func NewRecordingService(rc *redis.Client) *RecordingService {
+	return &RecordingService{rc: rc}
+}
+
+func (s *RecordingService) StartRecording(ctx context.Context, req *livekit.StartRecordingRequest) (*livekit.RecordingResponse, error) {
+	if err := EnsureAdminPermission(ctx, ""); err != nil {
```

Comment:
`roomAdmin` only gives them access to a single room, so we'd have to ensure room is matching?